web/src/components/Header.js

```diff
@@ -40,9 +36,14 @@ const Header = ({ isLoggedIn }) => {
 		return () => document.removeEventListener('click', handleDocumentClick)
 	})
 
-	useEffect(async () => {
-		const response = await getCategories()
-		setCategories(response)
+	useEffect(() => {
```

Comment:
The following code fixes the memory leak bug without using then. Can you check that this works?
```
useEffect( () => {
		let unmounted = false

		const getCats = async () => {
			const response = await getCategories()
			if(!unmounted) {
				setCategories(response)
			}
		}
		getCats()
		
		return () => unmounted = true
	}, [])
```

Reply:
Resolved. I changed it using this. It works (y)